pgjdbc/src/main/java/org/postgresql/core/v3/SimpleParameterList.java

```diff
@@ -210,6 +210,14 @@ public String toString(@Positive int index, boolean standardConformingStrings) {
           }
           return Double.toString(d);
 
+        case Oid.NUMERIC:
+          Number n = ByteConverter.numeric((byte[]) paramValue);
+          if (n instanceof Double) {
+            assert n.doubleValue() == Double.NaN;
```

Comment:
does this ever work?

Reply:
@vlsi, as in do we have any tests exercising that? Or if they did would this evaluate as expected?

Reply:
@vlsi I found some issues/inconsistencies with current handling of NaN and logged https://github.com/pgjdbc/pgjdbc/issues/1941.

Reply:
I think the reliably way to test for NaN is `assert n.isNan()`

Reply:
updated